src/contracts/cameligo/betting/main.mligo

```diff
@@ -0,0 +1,311 @@
+  let (newbets_team_two, newbets_team_two_index) : ((address, tez) map * nat) = match (Map.find_opt (Tezos.get_sender()) p_requested_event_id.bets_team_two) with
+    | Some prevAmount -> add_bet_team_two_amount_to_existing_user p_requested_event_id prevAmount
+    | None -> add_bet_team_twoAmountToNewUser p_requested_event_id
+  in
+  let newbets_team_two_total : tez = (p_requested_event_id.bets_team_two_total + Tezos.get_amount()) in
+  let r_updated_event : Types.event_bets = {p_requested_event_id with bets_team_two = newbets_team_two; bets_team_two_index = newbets_team_two_index; bets_team_two_total = newbets_team_two_total;} in
+  (r_updated_event)
+
+
+let add_bet (p_requested_event_id : nat)(team_one_bet : bool)(s : Types.storage) : (operation list * Types.storage) =
+  let _ = Assert.not_manager_nor_oracle (Tezos.get_sender()) s.manager s.oracle_address in
+  let _ = Assert.no_tez (Tezos.get_amount()) in
+  let _ = Assert.tez_lower_than_min (Tezos.get_amount()) s.bet_config.min_bet_amount in
+  let requested_event : Types.event_type = match (Big_map.find_opt p_requested_event_id s.events) with
+    | Some event -> event
+    | None -> failwith Errors.no_event_id
+  in
+  let _ = Assert.betting_not_finalized (requested_event.is_finalized) in
+  let _ = Assert.betting_before_period_start (requested_event.start_bet_time) in
+  let _ = Assert.betting_after_period_end (requested_event.closed_bet_time) in
+  let requested_event_bets : Types.event_bets = match (Big_map.find_opt p_requested_event_id s.events_bets) with
+    | Some event -> event
+    | None -> failwith Errors.no_event_bets
+  in
+  let updated_bet_event : Types.event_bets = if (team_one_bet)
+    then ( let uEvent : Types.event_bets = add_bet_team_one requested_event_bets in (uEvent) )
+    else ( let uEvent : Types.event_bets = add_bet_team_two requested_event_bets in (uEvent) )
+  in
+  let new_events_map : (nat, Types.event_bets) big_map = (Big_map.update p_requested_event_id (Some(updated_bet_event)) s.events_bets) in
+  (([] : operation list), {s with events_bets = new_events_map;})
+
+
+let make_transfer_op (addr : address ) ( value_won : tez ) (profit_quota : nat): operation =
+  let quota_to_send : nat = abs(100n - profit_quota) in
+  let value_to_send : tez = value_won * quota_to_send / 100n in
+  let dest_opt : unit contract option = Tezos.get_contract_opt addr in
+  let destination : unit contract = match dest_opt with
+    | None    -> failwith "Not found"
+    | Some ct -> ct
+  in
+  Tezos.transaction unit value_to_send destination
+
+let compose_paiement ( winner_map : (address, tez) map ) ( total_value_bet : tez ) ( total_value_won : tez ) (profit_quota : nat) : operation list =
+  let folded_op_list = fun (op_list, (address, bet_amount) : operation list * (address * tez) ) -> 
+    let won_reward_percentage : nat = 10000000n * bet_amount / total_value_bet in
+    let added_reward : tez =  total_value_won * won_reward_percentage / 10000000n in    
```

Comment:
I don't understand this calculation, Why `10000000n` ? Sorry if it is a dumb question

Reply:
Maybe there is a better way, but :
-> we should calculate the whole numerator, then the denominator, because tezos does not support the floating numbers
-> but we can't multiply tez by tez, so we can't begin with `total_value_won * bet_amount`
so we added the multiplication to keep the precision.

Reply:
Thanks for your explaination, I understood now